components/Banner/tests/unit/Theming/MDCBannerViewThemingTests.m

```diff
@@ -0,0 +1,219 @@
+  }
+
+  [self assertTraitCollectionAndElevationBlockForBannerView:self.bannerView
+                                                colorScheme:self.containerScheme.colorScheme];
+}
+
+- (void)testThemingWithCustomValues {
+  // Given
+  self.containerScheme.colorScheme.surfaceColor = UIColor.yellowColor;
+  self.containerScheme.colorScheme.onSurfaceColor = UIColor.cyanColor;
+  self.containerScheme.colorScheme.primaryColor = UIColor.purpleColor;
+  self.containerScheme.colorScheme.secondaryColor = UIColor.redColor;
+  self.containerScheme.colorScheme.backgroundColor = UIColor.blueColor;
+  self.containerScheme.colorScheme.elevationOverlayEnabledForDarkMode = YES;
+  self.containerScheme.typographyScheme.body2 = [UIFont systemFontOfSize:101.0];
+
+  // When
+  [self.bannerView applyThemeWithScheme:self.containerScheme];
+
+  // Then
+  NSUInteger maximumStateValue = UIControlStateNormal | UIControlStateSelected |
+                                 UIControlStateHighlighted | UIControlStateDisabled;
+  // Color
+  XCTAssertEqualObjects(self.bannerView.backgroundColor,
+                        self.containerScheme.colorScheme.surfaceColor);
+  XCTAssertEqualObjects(
+      self.bannerView.textView.textColor,
+      [self.containerScheme.colorScheme.onSurfaceColor colorWithAlphaComponent:kTextViewOpacity]);
+  XCTAssertEqualObjects(
+      self.bannerView.dividerColor,
+      [self.containerScheme.colorScheme.onSurfaceColor colorWithAlphaComponent:kDividerOpacity]);
+  for (NSUInteger state = 0; state <= maximumStateValue; ++state) {
+    XCTAssertEqualObjects([self.bannerView.leadingButton titleColorForState:UIControlStateNormal],
+                          self.containerScheme.colorScheme.primaryColor);
+    XCTAssertEqualObjects(
+        [self.bannerView.leadingButton imageTintColorForState:UIControlStateNormal],
+        self.containerScheme.colorScheme.primaryColor);
+    XCTAssertEqualObjects([self.bannerView.trailingButton titleColorForState:UIControlStateNormal],
+                          self.containerScheme.colorScheme.primaryColor);
+    XCTAssertEqualObjects(
+        [self.bannerView.trailingButton imageTintColorForState:UIControlStateNormal],
+        self.containerScheme.colorScheme.primaryColor);
+  }
+  // Typography
+  XCTAssertEqualObjects(self.bannerView.textView.font, self.containerScheme.typographyScheme.body2);
+
+  for (NSUInteger state = 0; state <= maximumStateValue; ++state) {
+    XCTAssertEqualObjects([self.bannerView.leadingButton titleFontForState:state],
+                          self.containerScheme.typographyScheme.button);
+    XCTAssertEqualObjects([self.bannerView.trailingButton titleFontForState:state],
+                          self.containerScheme.typographyScheme.button);
+  }
+
+  [self assertTraitCollectionAndElevationBlockForBannerView:self.bannerView
+                                                colorScheme:self.containerScheme.colorScheme];
+}
+
+- (void)testThemingWithPreScaledTypographyScheme {
```

Comment:
Nit/naming: The typography scheme here isn't pre-scaled, so the test name is a bit inaccurate. Perhaps `testFontsAreScaledWhenTypographySchemeRequestsPrescaling` is better?

Reply:
Addressed in3e3a6d981057498116cb359d873f2f82b97b8c93.